src/components/TldrawCanvasLabelDialog.tsx

```diff
@@ -306,15 +308,19 @@ const LabelDialog = ({
   const containerRef = useRef<HTMLDivElement>(null);
   const [error, setError] = useState("");
   const initialLabel = useMemo(() => {
-    if (_label) return _label;
+    if (_label)
+      return getPageTitleByPageUid(initialUid) || getTextByBlockUid(initialUid);
```

Comment:
why does truthiness of `_label` dictate to use data from initialUid instead?

Reply:
it's a proxy for "Is this a new canvas node or an existing canvas node?"

If it is existing, get most up to date node content.
If it's new, get the `specification` text instead

It's not used elsewhere so I renamed it to `isExistingCanvasNode`